src/js/component.js

```diff
@@ -1239,9 +1239,7 @@ class Component {
     fn = Fn.bind(this, fn);
 
     const timeoutId = window.setTimeout(fn, timeout);
-    const disposeFn = function() {
-      this.clearTimeout(timeoutId);
-    };
+    const disposeFn = () => this.clearTimeout(timeoutId);
```

Comment:
This seemed a bit weird to me in a regular function we wouldn't have access to `this.clearTimeout`

Reply:
we call the `dispose` handler with context as the component instance, so, it should have it.

Reply:
at least, it should be. this is ok too.